air/src/air/divisor.rs

```diff
@@ -54,10 +54,6 @@ impl<B: StarkField> ConstraintDivisor<B> {
     /// The above divisor specifies that transition constraints must hold on all steps of the
     /// execution trace except for the last $k$ steps.
     pub fn from_transition(trace_length: usize, num_exemptions: usize) -> Self {
-        assert!(
-            num_exemptions > 0,
-            "invalid number of transition exemptions: must be greater than zero"
-        );
         let exemptions = (trace_length - num_exemptions..trace_length)
             .map(|step| get_trace_domain_value_at::<B>(trace_length, step))
             .collect();
```

Comment:
This would re-introduce some Lagrange kernel concepts into the rest of the API (something we've been trying to avoid elsewhere). I also don't think this `assert!` is necessary, since `AirContext`'s constructor already checks it (e.g. for [main trace](https://github.com/plafer/winterfell/blob/9ea6c2df597e1fc110a93b6f94058de18a840b96/air/src/air/context.rs#L103)).